llvm/include/llvm/CodeGen/ValueTypes.td

```diff
@@ -34,16 +35,22 @@ class VTAny<int value> : ValueType<0, value> {
 
 class VTInt<int size, int value>
     : ValueType<size, value> {
+  let LLVMTyForEVT = !strconcat("Type::getIntNTy(Context, ",
+                                !cast<string>(size), ")");
   let isInteger = true;
 }
 
-class VTFP<int size, int value>
+class VTFP<int size, string LLVMTy, int value>
     : ValueType<size, value> {
+  let LLVMTyForEVT = !strconcat("Type::get", LLVMTy, "(Context)");
```

Comment:
I think it's probably more natural to just have tablegen directly emit the code for these, rather than splitting the code emission logic into strings in the td files and the emitter

Reply:
Yeah, make sense, thanks for the suggestion :)